### PR TITLE
Add UserInformationServlet with CRUD functionality

### PR DESCRIPTION
**Pull request by astrobliss**
### Description:
UserInformationServlet is responsible for Creation, Reading, Updating,
and Deleting User Objects. The User Objects are stored with DataStore,
and contain information about users who log on using UserService.

### Changes:
UserInformationServlet is implimented and active.
index.html now has a post field for interacting with
UserInformationServlet.

### Purpose:
This allows our app to have control of what is being stored indepndent
of the UserService. Currently its main purpose is to hold user
displayNames, in the future it could tie Users to profilePictures.

### Note:
Creation, Updates, and Deletion are only allowed if the User being acted
on is logged in.

### Security Concerns:
1) Anyone can Read anyone's data so long as they know their user-id
Not a huge concern, UserService isn't intended to hold sensitive
information.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Uses UserService and DataStore to allow Creation, Reading, and Updating of User objects
+ * JSON is sent on GET, on POST create a new user object or update individual parameters
+ */
```
Comment (chrisachille) [inline portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java:24]: As mentioned in your code comment, as by design, you have expectation that userInfo can be retrieved by anyone.

What do you think about making this point as a class comment, that user data is accessible by anyone by design, and it should not contain any private information.

File: portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  /**
+   * Creates or Updates the current user's User Object with parameters given
+   * If User isn't logged in return a HTTP 400 error code, Datastore is not modified
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if(!userService.isUserLoggedIn()) {
+      response.setStatus(400);
```
Comment (chrisachille) [inline portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java:87]: Can you return a 400 and redirect in the same response?

File: portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+    String userId = userService.getCurrentUser().getUserId();
+    Entity userEntity = getUserEntity(userId);
+
+    if(userEntity == null) {
+      // Create and store a new entity
+      userEntity = new Entity("User", userId);
+      String userEmail = userService.getCurrentUser().getEmail();
+      for(String propertyName: userPropertyNames) {
+        String propertyValue = getRequestParameter(propertyName, request, "");
```
Comment (chrisachille) [inline portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java:99]: nit: " : "

**Review by chrisachille — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java (hunk 2 above)
Comment (astrobliss) [reply portfolio/src/main/java/com/google/sps/servlets/UserInformationServlet.java:87]: I believe you can, the 400 response code is just extra information

**Comment by chrisachille** [comment]
Don't forget to resolve the last comment, and continue checking this in.

**Pull request merged**